services/core/DataMover/tests/test_datamover.py

```diff
@@ -0,0 +1,696 @@
+    global volttron_instance1, volttron_instance2
+    # print "Fixture volttron_instance"
+    # if volttron_instance1 is None:
+    volttron_instance1, volttron_instance2 = get_volttron_instances(2)
+
+
+# Fixture for setup and teardown of publish agent
+@pytest.fixture(scope="module",
+                params=['volttron_2', 'volttron_3'])
+def publish_agent(request, volttron_instances, forwarder):
+    global volttron_instance1, volttron_instance2
+    #print "Fixture publish_agent"
+    # 1: Start a fake agent to publish to message bus
+    if request.param == 'volttron_2':
+        agent = PublishMixin(
+            volttron_instance1.opts['publish_address'])
+    else:
+        agent = volttron_instance1.build_agent(identity='test-agent')
+
+    # 2: add a tear down method to stop sqlhistorian agent and the fake
+    # agent that published to message bus
+    def stop_agent():
+        print("In teardown method of publish_agent")
+        if isinstance(agent, Agent):
+            agent.core.stop()
+
+    request.addfinalizer(stop_agent)
+    return agent
+
+
+@pytest.fixture(scope="module")
+def query_agent(request, volttron_instances, sqlhistorian):
+    # print "Fixture query_agent"
+    # 1: Start a fake agent to query the sqlhistorian in volttron_instance2
+    agent = volttron_instance2.build_agent()
+
+    # 2: add a tear down method to stop sqlhistorian agent and the fake
+    # agent that published to message bus
+    def stop_agent():
+        print("In teardown method of module")
+        agent.core.stop()
+
+    request.addfinalizer(stop_agent)
+    return agent
+
+
+@pytest.fixture(scope="module")
+def sqlhistorian(request, volttron_instances):
+    # print "Fixture sqlhistorian"
+    global volttron_instance1, volttron_instance2
+    global sqlite_config
+    # 1: Install historian agent
+    # Install and start sqlhistorian agent in instance2
+    agent_uuid = volttron_instance2.install_agent(
+        agent_dir="services/core/SQLHistorian",
+        config_file=sqlite_config,
+        start=True,
+        vip_identity='platform.historian')
+    print("sqlite historian agent id: ", agent_uuid)
+
+
+
+@pytest.fixture(scope="module")
+def forwarder(request, volttron_instances):
+    #print "Fixture forwarder"
+    global volttron_instance1, volttron_instance2
+
+    global forwarder_uuid, forwarder_config
+    # 1. Update destination address in forwarder configuration
+
+    if volttron_instance1.encrypt:
+        tf = tempfile.NamedTemporaryFile()
+        ks = KeyStore(tf.name)
+        # generate public private key pair for instance1
+        ks.generate()
+
+        # add public key of instance1 to instance2 auth file
+        authfile = AuthFile(volttron_instance2.volttron_home + "/auth.json")
+        entry = AuthEntry(credentials=ks.public)
+        authfile.add(entry)
+
+        # setup destination address to include keys
+        forwarder_config["destination-vip"] =\
```

Comment:
yeah that's actually not a bad thing....if the keys are still in there and the tests pass...however we need to test both ways and they need to be mutually exclusive...

Must be all in url or none but serverkey specified.